host_core/lib/host_core/providers/provider_module.ex

```diff
@@ -113,6 +114,18 @@ defmodule HostCore.Providers.ProviderModule do
     {:noreply, state}
   end
 
+  def handle_info({:DOWN, _ref, :port, _port, :normal}, state) do
+    Logger.debug("Received DOWN message from port (executable stopped normally)")
+
+    {:noreply, state}
+  end
+
+  def handle_info({:DOWN, _ref, :port, _port, reason}, state) do
+    Logger.error("Received DOWN message from port (executable stopped) - #{reason}")
+    publish_provider_stopped(state.public_key, state.link_name, state.instance_id, "#{reason}")
+    {:stop, reason, state}
+  end
+
```

Comment:
Is there a reason to not publish that the provider stopped even if it stopped normally?

Reply:
Because if the provider stops normally, then there's another function that's been called that will publish the stop. We're not publishing the stop here because that would actually result in a double-publish of the event.

Reply:
Ah ok, so stopping with a reason of `:normal` means we essentially told that port to go DOWN, any other reason (like being killed by the OS, failing to startup) falls under the other handler. 👍🏻

Reply:
In other words, the only reason we'd ever get the `:normal` shutdown from the child port is because the parent process (the provider module) was told to terminate normally, in which case it publishes the provider stopped event.